src/memo/memo.ml

```diff
@@ -314,7 +314,7 @@ module Cache_lookup = struct
      outcome would have been [Out_of_date {old_value = None}] instead. *)
   module Failure = struct
     type 'a t =
-      | Out_of_date of { old_value : 'a option }
+      | Out_of_date of { old_value : 'a Option.Unboxed.t }
```

Comment:
Isn't it unsafe to use `Option.Unboxed` here because `'a` can be anything?

Reply:
"Unsafe" in this context would mean that you just get a runtime error. You will not get a segfault. In this particular case, it's alright because the old_value is always `Cached_value.t`. I could move the definition to the recursive module `M` to specialize this type, but I'm not quite a fan of that either. Do you have a preference?

Reply:
Got it. I think we need to add a comment explaining why it's safe to use `Option.Unboxed.t` here.

Reply:
Done.